lib/server.js

```diff
@@ -5,7 +5,11 @@ let DEFAULTS = {
 	port: 3000
 };
 
-exports.static = (config, webroot) => {
+exports.static = (config, webroot, devServer = {}) => {
+	if(devServer.middlewares) {
+		abort("--static doesn't support middlewares right now, use --liveserve");
+	}
```

Comment:
doesn't this mean that if we're working on the same project, I'd be forced to use live server rather than the sensible option?

Reply:
I understand (and suspected) that rationale, but given the consequences, this strikes me as _very_ icky. In part that's because such middleware might very well be a convenience rather than a functional requirement.

Of course I understand that making donny support middleware would be an extra burden (even if it was a trivial change, which these things rarely end up being), so I wasn't implying that more work there was the solution.

Indeed, it strikes me as inevitable that these middleware plugins will need to be aware of which server they're to be used with. (It's worth noting that this is, in fact, a new kind of plugin we're introducing here, with all the consequences WRT APIs.)